client/verify.go

```diff
@@ -4,170 +4,18 @@
-	}
-
-	// Convert the record to JSON.
-	recordJSON, err := json.Marshal(req.GetRecord())
-	if err != nil {
-		return nil, fmt.Errorf("failed to marshal record: %w", err)
-	}
-
-	oidcVerifier := req.GetProvider().GetOidc()
-
-	// Load identity verification options.
-	var identityPolicy verify.PolicyOption
-	{
-		// Create OIDC identity matcher for verification.
-		certID, err := verify.NewShortCertificateIdentity("", oidcVerifier.GetExpectedIssuer(), "", oidcVerifier.GetExpectedSigner())
-		if err != nil {
-			return nil, fmt.Errorf("failed to create certificate identity: %w", err)
-		}
-
-		identityPolicy = verify.WithCertificateIdentity(certID)
-	}
-
-	// Load trusted root material.
-	var trustedMaterial root.TrustedMaterialCollection
-	{
-		// Get staging TUF trusted root.
-		// TODO: allow switching between TUF environments.
-		fetcher := fetcher.NewDefaultFetcher()
-		fetcher.SetHTTPUserAgent(util.ConstructUserAgent())
-		tufOptions := &tuf.Options{
-			Root:              tuf.StagingRoot(),
-			RepositoryBaseURL: tuf.StagingMirror,
-			Fetcher:           fetcher,
-			DisableLocalCache: true, // read-only mode; prevent from pulling root CA to local dir
-		}
-
-		tufClient, err := tuf.New(tufOptions)
-		if err != nil {
-			return nil, fmt.Errorf("failed to create TUF client: %w", err)
-		}
-
-		trustedRoot, err := root.GetTrustedRoot(tufClient)
-		if err != nil {
-			return nil, fmt.Errorf("failed to get trusted root: %w", err)
-		}
-
-		trustedMaterial = append(trustedMaterial, trustedRoot)
-	}
-
-	// Create verifier session.
-	sev, err := verify.NewVerifier(trustedMaterial,
-		verify.WithSignedCertificateTimestamps(1),
-		verify.WithObserverTimestamps(1),
-		verify.WithTransparencyLog(1),
-	)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create verifier: %w", err)
-	}
-
-	// Run verification
-	_, err = sev.Verify(sigBundle, verify.NewPolicy(verify.WithArtifact(bytes.NewReader(recordJSON)), identityPolicy))
-	if err != nil {
-		return nil, fmt.Errorf("failed to verify signature: %w", err)
-	}
-
-	response := &signv1.VerifyResponse{
-		Success: err == nil,
-	}
-
-	// Verify the signature.
-	return response, nil
-}
-
-// VerifyWithKey verifies the signature of the record using a PEM-encoded public key.
-func (c *Client) VerifyWithKey(_ context.Context, req *signv1.VerifyRequest) (*signv1.VerifyResponse, error) {
-	keyVerifier := req.GetProvider().GetKey()
-
-	// Validate request.
-	if req.GetRecord() == nil {
-		return nil, errors.New("record must be set")
-	}
-
-	if req.GetSignature() == nil {
-		return nil, errors.New("signature must be set")
-	}
-
-	// Extract signature data.
-	sigBundleRawJSON, err := base64.StdEncoding.DecodeString(req.GetSignature().GetContentBundle())
+// VerifyWithZot verifies the signature of the record using zot's verification API via the server.
+func (c *Client) VerifyWithZot(ctx context.Context, req *signv1.VerifyRequest) (*signv1.VerifyResponse, error) {
```

Comment:
i would consider either renaming this to only `Verify` or if we want to have local and remote verification, then `VerifyRemote`
```suggestion
// VerifyRemote verifies the signature of the record on the the server.
func (c *Client) VerifyRemote(ctx context.Context, req *signv1.VerifyRequest) (*signv1.VerifyResponse, error) {
```